clientSecurity/src/main/java/org/apache/geode/examples/clientSecurity/Example.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.examples.clientSecurity;
+
+import static org.apache.geode.distributed.ConfigurationProperties.SECURITY_CLIENT_AUTH_INIT;
+
+import java.util.Properties;
+
+import org.apache.commons.lang.Validate;
+import org.apache.logging.log4j.Logger;
+
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.client.ClientCache;
+import org.apache.geode.cache.client.ClientCacheFactory;
+import org.apache.geode.cache.client.ClientRegionShortcut;
+import org.apache.geode.internal.logging.LogService;
+
+public class Example implements AutoCloseable {
+  private static final Logger logger = LogService.getLogger();
+
+  private static final String REGION1 = "region1";
+  private static final String REGION2 = "region2";
+
+  // Some example data
+  private static final String AUTHOR_GROSSMAN = "Grossman";
+  private static final String BOOK_BY_GROSSMAN = "Soon I Will Be Invincible";
+
+  private static final String AUTHOR_ROTHFUSS = "Rothfuss";
+  private static final String BOOK_BY_ROTHFUSS = "The Name of the Wind";
+
+  private static final String AUTHOR_LYNCH = "Lynch";
+  private static final String BOOK_BY_LYNCH = "The Lies of Locke Lamora";
+
+  private static final String AUTHOR_SCALZI = "Scalzi";
+  private static final String BOOK_BY_SCALZI = "Old Man's War";
+
+  private static final String AUTHOR_SANDERSON = "Sanderson";
+  private static final String BOOK_BY_SANDERSON = "The Way of Kings";
+
+  private static final String AUTHOR_ABERCROMBIE = "Abercrombie";
+  private static final String BOOK_BY_ABERCROMBIE = "The Blade Itself";
+
+  private final ClientCache cache;
+  private final Region<String, String> region1;
+  private final Region<String, String> region2;
+
+  private Example(String username, String password) {
+    Properties props = new Properties();
+    props.setProperty(ExampleAuthInit.USER_NAME, username);
+    props.setProperty(ExampleAuthInit.PASSWORD, password);
```

Comment:
I think it is confusing to put the credentials in this properties. The properties here should just specify the AuthInit. In AuthInit.getCredentials() the user is supposed to use their existing infrastructure (ldap or anything else) to obtain the credentials and put them in the properties.